### PR TITLE
Reference Flux Handler

### PR DESCRIPTION
**Pull request by guynir42**
Add a the input/output code for getting reference flux into the photometry table and out of it. 

Also added tests.

### CODE REVIEW

**Comment by guynir42** [comment]
@profjsb I added a test for the hybrid properties. This is ready for review.

File: skyportal/tests/api/test_photometry.py [hunk 1]
```diff
@@ -53,6 +55,404 @@ def test_token_user_post_get_photometry_data(
...
+    photometry_id = data['data']['ids'][0]
+    status, data = api(
+        'GET', f'photometry/{photometry_id}?format=both', token=upload_data_token
+    )
+    assert_api(status, data)
+
+    assert data['data']['altdata'] == {'some_key': 'some_value'}
+
+    expected_flux = 10 ** (-0.4 * (19.24 - 23.9))
+    expected_fluxerr = 0.123 / 2.5 * np.log(10) * expected_flux
```
Comment (profjsb) [inline skyportal/tests/api/test_photometry.py:233]: This happens to work but I think parentheses would make for a better read:
`0.123*(np.log(10)/2.5)*expected_flux`

File: skyportal/tests/api/test_photometry.py [hunk 2]
```diff
@@ -53,6 +55,404 @@ def test_token_user_post_get_photometry_data(
...
+    assert np.isclose(data['data']['flux'], expected_flux)
+    assert np.isclose(data['data']['fluxerr'], expected_fluxerr)
+    assert np.isclose(data['data']['mag'], 19.24)
+    assert np.isclose(data['data']['magerr'], 0.123)
+    assert np.isclose(data['data']['magref'], 17.01)
+    assert np.isclose(data['data']['e_magref'], 0.01)
+
+    expected_ref_flux = 10 ** (-0.4 * (17.01 - 23.9))
+    expected_ref_fluxerr = 0.01 / 2.5 * np.log(10) * expected_ref_flux
```
Comment (profjsb) [inline skyportal/tests/api/test_photometry.py:242]: same as above comment

File: skyportal/tests/api/test_photometry.py [hunk 3]
```diff
@@ -53,6 +55,404 @@ def test_token_user_post_get_photometry_data(
...
+        token=upload_data_token,
+    )
+    assert_api(status, data)
+    status, data = api(
+        'GET', f'photometry/{photometry_id}?format=both', token=upload_data_token
+    )
+    assert_api(status, data)
+
+    expected_ref_flux = 10 ** (-0.4 * (18.01 - 23.9))
+    expected_ref_fluxerr = 0.02 / 2.5 * np.log(10) * expected_ref_flux
```
Comment (profjsb) [inline skyportal/tests/api/test_photometry.py:288]: same

**Review by profjsb — CHANGES_REQUESTED** [review]
Looks good. Small comments to fix up.

File: skyportal/handlers/api/photometry.py [hunk 4]
```diff
@@ -115,6 +115,21 @@ def serialize(phot, outsys, format):
         'groups': phot.groups,
         'altdata': phot.altdata,
     }
+    if (
+        phot.ref_flux is not None
+        and not np.isnan(phot.ref_flux)
+        and phot.ref_fluxerr is not None
+        and not np.isnan(phot.ref_fluxerr)
+    ):
+        print(f'adding ref and tot fluxes. ref flux is {phot.ref_flux}')
```
[the remark this answers is not in the document]
Comment (guynir42) [reply skyportal/handlers/api/photometry.py:124]: Just left over from debugging.

**Pull request merged**